galsim/compound.py

```diff
@@ -1159,3 +1322,7 @@ def __repr__(self):
             gsparams=repr(self._input_gsparams),
         )
         return rep
+
+    def _prepareDraw(self):
+        # RandomWalk never wraps a PhaseScreenPSF, so no need to prepare anything.
+        pass
```

Comment:
It's because `RandomWalk` derives from `Sum`, which does have an override.  So this de-overrides that override.